jgrapht-core/src/main/java/org/jgrapht/GraphTests.java

```diff
@@ -479,6 +479,32 @@
         return new HierholzerEulerianCycle<V, E>().isEulerian(graph);
     }
 
+    /**
+     * Tests whether a graph meets Ore's condition to be Hamiltonian.
+     *
+     * Let G be a (finite and simple) graph with n ≥ 3 vertices. We denote by deg v the degree of a vertex v in G, i.e. the number of incident edges in G to v.
+     * Then, Ore's theorem states that if deg v + deg w ≥ n for every pair of distinct non-adjacent vertices v and w of G, then G is Hamiltonian.
+     *
+     * @param graph the input graph
+     * @param <V> the graph vertex type
+     * @param <E> the graph edge type
+     * @return true if the graph meets Ore's condition, false otherwise
+     * @see org.jgrapht.alg.tour.PalmerHamiltonianCycle
+     */
+    public static <V, E> boolean hasOreProperty(Graph<V, E> graph){
```

Comment:
You need to specify here (in the docs) and perhaps check in the implementation that the graph is undirected, etc.

Reply:
I'm not sure I have to. I mention that the graph has to be finite and simple so that implies undirected.

Reply:
Not everyone understands this immediately. Anyway, since we added the graph type information, we check some preconditions on runtime. See `graph.getType()`.

Reply:
I have updated the docs. Now it's clear that it's undirected.

Reply:
The most important thing here is to add a run-time check:

```
requireUndirected(graph);
````

Reply:
I have added it but I still don't understand something: isn't some checking of undirected-ness also done in `isSimple`?